### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by zimnx**
Updates dependencies to newest versions.

Requires #29 to pass CI.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -5,53 +5,54 @@ go 1.21
 require (
 	github.com/container-storage-interface/spec v1.9.0
 	github.com/gocql/gocql v1.6.0
```
Comment (tnozicka) [inline go.mod:7]: should we bump it as well?

File: go.mod [hunk 2]
```diff
@@ -61,85 +62,83 @@ require (
...
+	google.golang.org/genproto/googleapis/api v0.0.0-20231106174013-bbf56f31fb17 // indirect
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20231106174013-bbf56f31fb17 // indirect
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/natefinch/lumberjack.v2 v2.2.1 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
-	k8s.io/apiextensions-apiserver v0.0.0 // indirect
-	k8s.io/cloud-provider v0.0.0 // indirect
+	k8s.io/apiextensions-apiserver v0.28.3 // indirect
+	k8s.io/cloud-provider v0.28.3 // indirect
```
Comment (tnozicka) [inline go.mod:122]: v0.28.4 now

Comment (tnozicka) [reply go.mod:122]: https://github.com/scylladb/k8s-local-volume-provisioner/security/dependabot/15

**Comment by tnozicka** [comment]
otherwise this look in line with https://github.com/kubernetes/kubernetes/pull/121842 so we can bump ahead of time

File: go.mod (hunk 1 above)
Comment (zimnx) [reply go.mod:7]: This is the latest upstream one. We should get rid of it anyway.

File: go.mod (hunk 2 above)
Comment (zimnx) [reply go.mod]: 0.28.4 wasn't available when this PR was created. Updated it.

**Review by tnozicka — APPROVED** [review]
/approve
/lgtm
thanks

**Pull request merged**